src/harness/reference_models/iap/iap.py

```diff
@@ -0,0 +1,529 @@
+    esc_antenna_info: contains information on ESC antenna height, azimuth, 
+                      gain and pattern gain
+    threshold: protection threshold (mW)
+    protection_ent_type: an enum member of class ProtectedEntityType
+  Returns:
+    Updates asas_interference instance of class type AggregateInterferenceOutputFormat.
+    It is updated in the format of {latitude : {longitude : [ asas_p(mW/IAPBW),.. ]}}
+    and aggregate_interference instance of class type AggregateInterferenceOutputFormat.
+    It is updated in the format of {latitude : { longitude : [a_p(mW/IAPBW),..] }}
+  """
+  
+  # Get all the grants inside neighborhood of the protection entity
+  grants_inside = interf.findGrantsInsideNeighborhood(
+                    grant_objects, protection_point, protection_ent_type)
+    
+  if len(grants_inside) > 0:
+
+    for channel in channels:
+      # Set interference quota for protection_point, channel combination
+
+      # Calculate Roll of Attenuation for ESC Sensor Protection entity
+      if protection_ent_type is interf.ProtectedEntityType.ESC:
+        if channel[0] >= 3650.e6:
+          center_freq = (channel[0] + channel[1]) / 2
+          interference_quota = interf.dbToLinear(interf.linearToDb(threshold) - 
+                (2.5 + ((center_freq - interf.ESC_CH21_CF_HZ) / interf.MHZ)))
+        else:
+          interference_quota = threshold
+
+      # Get the FSS blocking passband range
+      elif protection_ent_type is interf.ProtectedEntityType.FSS_BLOCKING:
+        interference_quota = threshold
+        channel = [low_freq, high_freq]
+
+      else:
+        interference_quota = threshold
+
+      # Get protection constraint over 5MHz channel range
+      channel_constraint = interf.ProtectionConstraint(latitude=protection_point[0],
+                             longitude=protection_point[1], low_frequency=channel[0],
+                             high_frequency=channel[1], entity_type=protection_ent_type)
+
+      # Identify CBSD grants overlapping with frequency range of the protection entity 
+      # in the neighborhood of the protection point and channel
+      neighborhood_grants = interf.findOverlappingGrants(
+                              grants_inside, channel_constraint)
+      num_unsatisfied_grants = len(neighborhood_grants)
+
+      if not neighborhood_grants:
+        continue
+
+      # calculate the fair share
+      fair_share = float(interference_quota) / num_unsatisfied_grants 
+
+      # Empty list of final IAP EIRP and interference assigned to the grant
+      grants_eirp = []
+
+      aggr_interference = 0
+
+      a_sas_p_interference = 0
+
+      # Initialize flag to False for all the grants
+      grant_initialize_flag = [False] * num_unsatisfied_grants 
+
+      # Initialize grant EIRP to maxEirp for all the grants 
+      grants_eirp = [grant.max_eirp for grant in neighborhood_grants]
+
+      while (num_unsatisfied_grants > 0):
```

Comment:
Please review again this logic which not quite right I think: see below.

Reply:
Moved   fairshare calculation outside for loop and set the grant_initialize_flag to   True for all the satisfied grants. These grants will not be considered for   future iterations